File: thesis/algorithms/java/Np.java
import java.io.*;
import java.nio.file.*;

public class Np {
    public static boolean hasFactor(int n) {
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) return true;
        }
        return false;
    }

    public static void main(String[] args) {
        String tamanho = args.length > 0 ? args[0] : "small";
        String datasetPath = "datasets/" + tamanho + "/factoring.json";

        try {
            String content = new String(Files.readAllBytes(Paths.get(datasetPath))).trim();
            // Remove colchetes e espaços extras
            content = content.replace("[", "").replace("]", "").replaceAll("\\s", "");

            if (content.isEmpty()) {
                System.out.printf("Nenhum número no dataset %s.%n", tamanho);
                return;
            }

            String[] parts = content.split(",");
            int count = 0;
            for (String part : parts) {
                int n = Integer.parseInt(part);
                if (hasFactor(n)) count++;
            }

            System.out.printf("Fatorados %d números do dataset %s.%n", count, tamanho);

        } catch (IOException e) {
            System.out.println("Erro ao ler o arquivo: " + e.getMessage());
        } catch (NumberFormatException e) {
            System.out.println("Erro ao converter número: " + e.getMessage());
        }
    }
}

File: thesis/algorithms/c#/np.cs
using System;
using System.IO;
using System.Collections.Generic;

class Np
{
    static void Main(string[] args)
    {
        string tamanho = args.Length > 0 ? args[0] : "small";
        string path = $"datasets/{tamanho}/factoring.json";

        if (!File.Exists(path))
        {
            Console.WriteLine($"Arquivo não encontrado: {path}");
            return;
        }

        string json = File.ReadAllText(path);
        json = json.Trim().TrimStart('[').TrimEnd(']');
        string[] parts = json.Split(',');

        int count = 0;
        foreach (string part in parts)
        {
            if (int.TryParse(part.Trim(), out int n))
            {
                for (int i = 2; i * i <= n; i++)
                {
                    if (n % i == 0)
                    {
                        count++;
                        break;
                    }
                }
            }
        }

        Console.WriteLine($"Fatorados {count} números do dataset {tamanho}.");
    }
}
